### PR TITLE
Fix PostCSS plugin ignoring .stylelintignore

### LINKED ISSUE #2949: postcss ignores .stylelintignore
Here's my webpack snippet:
```
    {
        test: /\.css$/,
        use: [
          require.resolve('style-loader'),
          {
            loader: require.resolve('css-loader'),
            options: {
              importLoaders: 1
            },
          },
          {
            loader: require.resolve('postcss-loader'),
            options: {
              plugins: () => [
                require('postcss-import'),
                require('stylelint')(),
                require("postcss-reporter")(),
              ]
            },
          },
        ],
      },
```

I have `.stylelintignore` located next to `package.json`. 
CLI rescpects `.stylelintignore` ( command `stylelint "**/*.css"`), but postss ignores it.
Is this my mistake or is this a bug?

### LINKED ISSUE #4134: Fix PostCSS plugin ignoring .stylelintignore
> Clearly describe the bug

This is an exist issue，although it was closed. https://github.com/stylelint/stylelint/issues/2949

For example, in postcss configuration:

```javascript
module.exports = {
	plugins: [
		require('postcss-import')({
			plugins: [
				require('stylelint')({})
			]
		}),
```
stylelint did not resolve ignore file.

> Which version of stylelint are you using?

```"stylelint": "^9.2.0",```

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Use `stylelint` Programmatically. PostCSS plugin and Gulp plugin.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

Stylelint should resolve ignore file.

> What actually happened (e.g. what warnings or errors did you get)?

Ignore configuration in `.stylelintignore` did not works.

### PR DESCRIPTION
**Pull request by joe223**
> Which issue, if any, is this issue related to?

Closes #4134.

> Is there anything in the PR that needs further explanation?

Relative documentation https://github.com/joe223/stylelint/blob/master/docs/user-guide/postcss-plugin.md#ignorepath.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ node_modules
 .eslintcache
 package-lock.json
 yarn.lock
+.idea
```
Comment (hudochenkov) [inline .gitignore]: This shouldn't be in a project gitignore. This belongs to user's [global gitignore](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore).

Comment (joe223) [reply .gitignore]: 👌,
And there still some unit test error to be resolved.

**Comment by joe223** [comment]
@hudochenkov Unit test all passed.

**Comment by joe223** [comment]
I have checked the `https://github.com/stylelint/stylelint/blob/49bed2c87b274e2684bad71e0fb709500f114765/lib/__tests__/stylelintignore-test/stylelintignore.test.js#L20`.

This unit tests cover standalone's ignore configuration. I guess PostCSS plugin API ignorePath configuration unit test should be almost.

**Comment by joe223** [comment]
@hudochenkov I have add one unit test for postcssPlugin `ignorePath` option, this option works with `.stylelintignore` file to ignore style files.

**Comment by hudochenkov** [comment]
Thank you!

**Comment by hudochenkov** [comment]
> -   Fixed: PostCSS plugin ignoring .stylelintignore ([#4186](https://github.com/stylelint/stylelint/pull/4186)).

**Pull request merged**